ecosystem/sep-0034.md

```diff
@@ -0,0 +1,120 @@
+## Preamble
+
+```
+SEP: 0034
+Title: Wallet Request Attribution for Anchors
+Author: Jake Urban, Leigh McCulloch
+Track: Standard
+Status: Draft
+Created: 2020-08-17
+Discussion: https://github.com/stellar/stellar-protocol/issues/685
+```
+
+## Simple Summary
+This protocol defines how anchors can cryptographically attribute HTTPS requests to wallet applications. 
+
+## Motivation
+In general, web services may want to attribute resources created or actions taken on their service to client applications. Because requests headers like `User-Agent` can be spoofed, these web services require a cryptographically secure method of verifying a client's identity.
+
+Throughout this document, the resource mentioned is a deposit or withdrawal transaction, but the resource or action that requires identity verification could be related to different use cases between wallets and anchors.
+
+## Abstract
+This following protocol involves 3 parties:
+
+- The Wallet: the application that makes HTTPS requests to the anchor server for a resource or action
+- The Wallet Server: the web server that claims ownership of the wallet
+    - Note that frontend-only wallets _must_ add a backend web server to comply with this protocol
+- The Anchor Server: the web server that verifies the wallet's claims
+
+At a high level, the Wallet Server generates a [JSON Web Signature](https://www.rfc-editor.org/rfc/rfc7515.html) (JWS) that the Wallet receives and includes in future requests to the Anchor Server. The Anchor Server verifies that the JWS included in the request was signed by the Wallet Server and that the transaction (or resource) ID was included in the decoded [JWT](https://tools.ietf.org/html/rfc7519) payload. These steps prove that the Wallet Server takes ownership of the requests being made for a particular transaction.
```

Comment:
could JWT be defined as JWS is, i.e. JSON Web Token (JWT)

Reply:
Yes, good idea